neuro_extras/image.py

```diff
@@ -240,42 +319,63 @@ async def _build_image(
     force_overwrite: bool,
     preset: Optional[str] = None,
     registry_auths: Sequence[DockerConfigAuth] = (),
+    local: bool = False,
     verbose: bool = False,
 ) -> int:
     async with get_neuro_client() as client:
         cluster = _get_cluster_from_uri(client, image_uri_str, scheme="image")
     async with get_neuro_client(cluster=cluster) as client:
         context_uri = client.parse.str_to_uri(
             context,
-            allowed_schemes=("file", "storage"),
+            allowed_schemes=("file",) if local else ("file", "storage"),
         )
         image_exists = await _check_image_exists(image_uri_str, client)
-        if image_exists and force_overwrite:
-            logger.warning(
-                f"Target image '{image_uri_str}' exists and will be overwritten."
-            )
-        elif image_exists and not force_overwrite:
-            raise click.ClickException(
-                f"Target image '{image_uri_str}' exists. "
-                f"Use -F/--force-overwrite flag to enforce overwriting."
-            )
+        if image_exists:
+            if force_overwrite:
+                logger.warning(
+                    f"Target image '{image_uri_str}' exists and will be overwritten."
+                )
+            else:
+                raise click.ClickException(
+                    f"Target image '{image_uri_str}' exists. "
+                    f"Use -F/--force-overwrite flag to enforce overwriting."
+                )
 
         builder = ImageBuilder(
```

Comment:
We might also apply smth like strategy pattern and rename the current ImageBuilder to RemoteImageBuilder and create LocalImageBuilder. Also, pushing the image logic implementation might be there.